programs/ziti-edge-tunnel/ziti-edge-tunnel.c

```diff
@@ -1129,6 +1130,19 @@ static void on_event(const base_event *ev) {
     }
 }
 
+static char* normalize_host(char* hostname) {
+    size_t len = strlen(hostname);
+    char* hostname_new = calloc(len+1, sizeof(char));
+    // add . in the beginning of the hostname
+    if (hostname[len-1] == ".") {
+        // remove the . from the end of the hostname
+        snprintf(hostname_new, len * sizeof(char), ".%s", hostname);
+    } else {
+        sprintf(hostname_new,".%s", hostname);
```

Comment:
This creates an invalid memory access at the end of `hostname_new` where the terminating '\0' should be.

`hostname_new` has strlen(hostname)+1 bytes; the extra byte gives you a place for the terminating nut byte. If you also want to add a character to the beginning of the string then you need to allocate `strlen(hostname)+2`.